Packages/vcs/Lib/VTKPlots.py

```diff
@@ -621,35 +626,10 @@ def plotVector(self,data1,data2,tmpl,gm,vtk_backend_grid=None,vtk_backend_geo=No
     returned["vtk_backend_grid"]=vtk_backend_grid
     returned["vtk_backend_geo"]=geo
     missingMapper = vcs2vtk.putMaskOnVTKGrid(data1,vtk_backend_grid,None,False,deep=False)
-    returned["vtk_backend_missing_mapper"]=missingMapper
-
-    u=numpy.ma.ravel(data1)
-    v=numpy.ma.ravel(data2)
-    sh = list(u.shape)
-    sh.append(1)
-    u = numpy.reshape(u,sh)
-    v = numpy.reshape(v,sh)
-    z = numpy.zeros(u.shape)
-    w = numpy.concatenate((u,v),axis=1)
-    w = numpy.concatenate((w,z),axis=1)
-
-    # HACK The grid returned by vtk2vcs.genGrid is not the same size as the
-    # data array. I'm not sure where the issue is...for now let's just zero-pad
-    # data array so that we can at least test rendering until Charles gets
-    # back from vacation:
-    wLen = len(w)
-    numPts = vtk_backend_grid.GetNumberOfPoints()
-    if wLen != numPts:
-        warnings.warn("!!! Warning during vector plotting: Number of points does not "\
-              "match the number of vectors to be glyphed (%s points vs %s "\
-              "vectors). The vectors will be padded/truncated to match for "\
-              "rendering purposes, but the resulting image should not be "\
-              "trusted."%(numPts, wLen))
-        newShape = (numPts,) + w.shape[1:]
-        w = numpy.ma.resize(w, newShape)
-
-    w = vcs2vtk.numpy_to_vtk_wrapper(w,deep=False)
-    w.SetName("vectors")
+    returned["vtk_backend_missing_mapper"]=missingMapper,None,False
```

Comment:
None is the color for missing, false is for cellData (True/False), these are the args sent before/later to: vcs2vtk.putMaskOnVTKGrid